### DISCUSSION TITLE
consecutive - concurrent collection.getFullList() doesn't emit in the front

### DISCUSSION THREAD

**Discussion by mathieuher** [pocketbase/pocketbase#3593] [Q&A] [answered]
Hey, thank again for PB.

Here is an example of TS code I use to get Object list and for each of them load an history (from another collection).

object.service.ts
```
// Get object from objects collection
public getSellingObjects$(): Observable<Object[]> {
        return from(environment.pb.collection('objects').getFullList()).pipe(
            map(objectRecords => objectRecords.map(objectRecord => new Object(objectRecord))),
            mergeAll(),
            switchMap(object => this.getObjectHistory$(object).pipe(
                map(history => {
                    object.history = history;
                    return object;
                })
            )),
            toArray()
          );
    }
 
// Get object history 
public getObjectHistory$(object: Object): Observable<ObjectHistory[]> {
        console.log('call history for object : ', object.id);
        return from(environment.pb.collection('objects_history').getFullList({ filter: `object = '${object.id}'`, $autoCancel: false })).pipe(
            tap(response => console.warn('History response : ', response),
            map(historyRecords => historyRecords.map(historyRecord => new ObjectHistory(historyRecord)))
        );
    }
```

On my test I've got 2 objects and for every of them I can see a GET on the objects_history collection : 
![image](https://github.com/pocketbase/pocketbase/assets/32459394/422510ec-ce7a-460f-a2b3-a3f7d8a2c76a)

But on the front-end side, only one history result is logged : 
![image](https://github.com/pocketbase/pocketbase/assets/32459394/4d3513ed-54ef-49f5-81c2-5cf18a1652e5)

I tried whitout filtering (objects_history collection have multiple records) but still the same problem.

Am I doing something wrong or is there an effective (known or not) issue here ?

PS : I didn't use inverse call and use expand relation because on some case I wan't to fine tune provided information and manage rights about related collection 

Thank you in advance for your response and your work !

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think this is PocketBase or JS SDK related.

I suspect that the issue is something with the client-side handling but there are not enough details to understand what those custom wrapped function are (is this RxJS?) and what is your data structure.

If you are not able to identify the cause of your issue, feel free to create a **minimal** reproducible repo (_ideally with export of your collections configuration from Admin UI > Settings > Export collections_) and I'll have a look.

**Reply by mathieuher** [reply]
Thank for the quick answer.
Sorry for the lack of informations, yes it's rxjs.

I will try to use an http call on the api to exclude JS SDK, and keep you informed.

**Reply by mathieuher** [reply]
Yep error was on my side by using wrong rxjs operator.

I should use mergeMap instead of switchMap for possible concurrent call to get objects_history 😄 
The first call on the objects_history was canceled by the second one.

Thank again for the great product !
